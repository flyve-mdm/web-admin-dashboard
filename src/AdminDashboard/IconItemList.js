import React from 'react'

export default class IconItemList extends React.Component {
    render() {
        let size = this.props.size
        return (
            <div
                className="profilePicture"
                style={{
                    backgroundColor: '#BBBBBB',
                    width: size,
                    height: size,
                    WebkitBorderRadius: size,
                    MozBorderRadius: size,
                    borderRadius: size,
                    backgroundSize: 'cover',
                    display: 'inline-block'
                }}
            >
                <img src="images/" height={size} width={size} alt="" />
            </div>
        )
    }
}
