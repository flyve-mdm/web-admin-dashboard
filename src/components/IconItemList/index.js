import React from 'react'
import PropTypes from 'prop-types'

export default class IconItemList extends React.Component {
    render() {
        const image = (this.props.type === 'file' && this.props.image !== '') ? require(`../../assets/images/${this.props.image}`) : this.props.image
        let style = {
            backgroundColor: this.props.backgroundColor,
            width: this.props.size,
            height: this.props.size,
            backgroundSize: 'cover',
            display: 'inline-block'
        }
        let className = ''
        if (this.props.type !== 'base64') {
            className = 'contentPicture'
            style = {
                ...style,
                WebkitBorderRadius: this.props.size,
                MozBorderRadius: this.props.size,
                borderRadius: this.props.size
            }
        }

        return (
            <div className={className} style={style}>
                <div className={this.props.imgClass} >
                    <img alt="" src={image} style={style} onClick={this.props.imgClick}/>
                </div>
            </div>
        )
    }
}

IconItemList.defaultProps = {
    size: 100,
    backgroundColor: '#e6e6e6',
    image: '',
    type: 'file'
}

IconItemList.propTypes = {
    size: PropTypes.number.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    type: PropTypes.oneOf(["file", "base64", "localFile"]).isRequired,
    imgClick: PropTypes.func,
    imgClass: PropTypes.string
}
