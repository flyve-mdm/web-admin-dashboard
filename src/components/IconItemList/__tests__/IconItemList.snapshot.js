import React from 'react'
import renderer from 'react-test-renderer'

import IconItemList from "../index"

describe('IconItemList', () => {
    test('renders IconItemList', () => {
        const component = renderer.create(
            <IconItemList image="profile.png"/>
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})