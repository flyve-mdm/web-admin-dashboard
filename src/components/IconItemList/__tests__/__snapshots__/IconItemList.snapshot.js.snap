// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconItemList renders IconItemList 1`] = `
<div
  className="list-pane__picture"
  style={
    Object {
      "MozBorderRadius": 100,
      "WebkitBorderRadius": 100,
      "backgroundColor": "#e6e6e6",
      "backgroundSize": "cover",
      "borderRadius": 100,
      "display": "inline-block",
      "height": 100,
      "width": 100,
    }
  }
>
  <div
    className=""
    onClick={[Function]}
    role="presentation"
  >
    <img
      alt=""
      src="profile.png"
      style={
        Object {
          "MozBorderRadius": 100,
          "WebkitBorderRadius": 100,
          "backgroundColor": "#e6e6e6",
          "backgroundSize": "cover",
          "borderRadius": 100,
          "display": "inline-block",
          "height": 100,
          "width": 100,
        }
      }
    />
  </div>
</div>
`;
